test/alternator/test_compressed_response.py

```
@@ -204,6 +204,48 @@ def assert_chunked(response_dict, **kwargs):
+# Whether a given response triggers the bug depends on the exact total number
+# of bytes flowing through the compressor. Experimentally, it recurs roughly
+# every 17KB, in a ~700-byte-wide window each time. We can't predict the
+# exact byte count once JSON/response overhead is added on top of the item
+# data, so below we sweep the size of a large item over a range wider than
+# one such period, finely enough that we can't miss the vulnerable window.
+def test_compressed_response_streamed_backlog(dynamodb, test_table_s):
+    # The bug reproduced here is in the shared compression code, not in
+    # anything specific to gzip or deflate so there's no need to check both.
+    encoding = "gzip"
+    with client_response_decompression(dynamodb):
+        with server_response_compression(dynamodb):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)